app/src/main/java/com/fastlink/zekrapp/ui/zekirScreen/ZekirScreenViewModel.kt

```diff
@@ -1,27 +1,45 @@
-package com.fastlink.zekrapp.viewModel
+package com.fastlink.zekrapp.ui.zekirScreen
 
+import androidx.compose.runtime.MutableState
 import androidx.compose.runtime.mutableIntStateOf
+import androidx.compose.runtime.mutableStateOf
+import androidx.lifecycle.SavedStateHandle
 import androidx.lifecycle.ViewModel
 import androidx.lifecycle.viewModelScope
-import com.fastlink.zekrapp.appData.ZekirSingleton
+import com.fastlink.zekrapp.appData.model.ZekirCategoryModel
+import com.fastlink.zekrapp.di.ZekirSingleton
 import com.fastlink.zekrapp.appData.model.ZekirModel
+import com.fastlink.zekrapp.di.ZekirCategorySingleton
 import dagger.hilt.android.lifecycle.HiltViewModel
 import kotlinx.coroutines.delay
 import kotlinx.coroutines.launch
 import javax.inject.Inject
 
 @HiltViewModel
-class ZekirViewModel @Inject constructor(
-    private val zekirSingleton: ZekirSingleton
+class ZekirScreenViewModel @Inject constructor(
+    private val zekirSingleton: ZekirSingleton,
+    private val zekirCategorySingleton: ZekirCategorySingleton,
+    savedStateHandle: SavedStateHandle,
 ) : ViewModel() {
-    private var categoryId: Int = 0
+
     val zekirCounter = mutableIntStateOf(0)
     var zekirs: List<ZekirModel> = emptyList()
         private set
+    var zekirCategory: MutableState<ZekirCategoryModel?> = mutableStateOf(null)
```

Comment:
Why are you making it null, this is unnecessary and makes crashes easier to happen.

Reply:
Initially, I used the lateinit keyword, but it caused the app to crash. So, I changed the type to accept null values to prevent crashes. Now, I have modified the code to this:

` var zekirCategory: MutableState<ZekirCategoryModel> =
        mutableStateOf(ZekirCategoryModel(1, "", ""))`

Is this better, or should I change it again?

Reply:
Keep it like this but add default values for the variables in the data class instead of adding values here.